modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/outgoing/OutgoingSnapshotsManager.java

```diff
@@ -141,8 +201,48 @@ private CompletableFuture<Void> respond(
             NetworkAddress sender,
             Long correlationId
     ) {
-        //TODO https://issues.apache.org/jira/browse/IGNITE-17262
+        //TODO https://issues.apache.org/jira/browse/IGNITE-17935
         // Handle offline sender and stopped manager.
         return messagingService.respond(sender, response, correlationId);
     }
+
+    @Override
+    public PartitionSnapshots partitionSnapshots(PartitionKey partitionKey) {
```

Comment:
Yes, it should return something non-nullable so that the one who makes a write needs to take a lock on the collection of snapshots of the partition.

I planned to clean it on partition destroy. For now, it is a handful of objects per alive partition, doesn't seem too wasteful.